### ISSUE TITLE
Build 170402 - No Deep Sleep (G920F)

### ISSUE THREAD

**Issue by karvek** [nexus7420/management#22]
I am on SM-G920F and my phone discharged of 22% in the last 2h 30m (9%/h) after a full charge.

According to GSam stats I have 2h 02m 58s wakelock time in the last 2h 30 power on time.
Doze Time: Light 1h 18m 39s (64%), Deep 0s (0%).
App consumption: Kernel 22.2%, Android 12.8%, NFC 9.1%

I think the phone is unable to reach deep sleep.
The issue is present without having any user app (a part from GSam) in background.

Maybe I should mention that my fingerprint hardware is not working (already in Touchwiz).

Build 2017-04-02

**Comment by Lukas0610** [comment]
It would help a much if you could post full battery-stats

**Comment by karvek** [comment]
The highest amount of wakelocks come from this entry:
_Combined App Wakelocks (non-Kernel)
WakeLock:174
Durata (inclusa attività):2h 39m 6s_

![screenshot 04 apr 2017 23_14_27](https://cloud.githubusercontent.com/assets/26905269/24679040/421e14f6-198d-11e7-9570-63dad99bea40.png)

![screenshot 04 apr 2017 23_22_15](https://cloud.githubusercontent.com/assets/26905269/24679507/9dbc31d0-198d-11e7-89ed-d73f7550f9c0.png)

Please be more specific if I have to do some particular test or bring some specific battery stats.
I am not sure how I can export the full battery stats with GSam. In former times I was using BetterBatteryStats but I am not sure if it is compatible with Nougat.

I can also flash again the ROM after a full wipe, but I remember that the drain was already there.

**Comment by karvek** [comment]
Ok, more informations.
I did a full wipe, flashed the same build+gapps pico+root and installed BetterBatteryStats.

A part from this nothing else.

I see that the **vfsspi_wake_lock** is keeping the phone 100% on.
With a fast search it looks like that this is related to the fingerprint sensor.
As I mentioned in the first post, the fingerprint hardware of my S6 is fucked up.

I then performed a new full wipe and flashed again lineage+gapps+supersu.
This time when asked in the initialization to set a fingerprint I denied it and now this specific wakelock disappeared.

I am not a dev, so sorry if I make it easy, but I would consider implementing something to avoid that the 'fingerprint initialization' gets stuck.

Now the issue is with NFC, in particular with **NfcService:mRoutingWakeLock**.

I updated my BatteryStats log for more quantitative feedback.
[BetterBatteryStats-2017-04-05_154207030.txt](https://github.com/nexus7420/management/files/896413/BetterBatteryStats-2017-04-05_154207030.txt)
[logcat-2017-04-05_154207138.txt](https://github.com/nexus7420/management/files/896412/logcat-2017-04-05_154207138.txt)

The NFC capability is grayed in the settings so I don't really know what to do here.

By the way thanks for keeping the project running.
